### PR TITLE
Small code readability improvements and more compact code

### PR DESCRIPTION
**Pull request by zntfdr**
Hi @younatics , thank you so much for making and sharing this nice framework.

While going throughout the code I've spotted a few lines that could have used some adjustments: this PR addresses most of them!

No changes have been made (except for the addition of a ``setLabelFont(_:)`` commodity function), the framework behaves exactly like before. 

Again, thank you so much for making this 😊

Cheers!

### CODE REVIEW

File: YNDropDownMenu/Source/YNDropDownMenu.swift [hunk 1]
```diff
@@ -35,11 +35,7 @@ open class YNDropDownMenu: UIView, YNDropDownDelegate {
         }
         set {
             guard let _dropDownViews = newValue else { return }
-            for view in _dropDownViews {
-                if let v = view as? YNDropDownView {
-                    v.delegate = self
-                }
-            }
+            _dropDownViews.flatMap({ $0 as? YNDropDownView }).forEach { $0.delegate = self }
```
Comment (younatics) [inline YNDropDownMenu/Source/YNDropDownMenu.swift:38]: Great point that you use closure 👍

File: YNDropDownMenu/Example/ViewController.swift [hunk 2]
```diff
@@ -19,15 +19,15 @@ class ViewController: UIViewController {
             // Inherit YNDropDownView if you want to hideMenu in your dropDownViews
             let view = YNDropDownMenu(frame: CGRect(x: 0, y: 64, width: UIScreen.main.bounds.size.width, height: 38), dropDownViews: _ZBdropDownViews, dropDownViewTitles: ["Apple", "Banana", "Kiwi", "Pear"])
             view.setImageWhen(normal: UIImage(named: "arrow_nor"), selected: UIImage(named: "arrow_sel"), disabled: UIImage(named: "arrow_dim"))
```
Comment (younatics) [inline YNDropDownMenu/Example/ViewController.swift]: Great, However I have to change menu label font to bold if menu is selected 👍

File: YNDropDownMenu/Example/ViewController.swift [hunk 3]
```diff
@@ -19,15 +19,15 @@ class ViewController: UIViewController {
...
-            view.setLabelColorWhen(normal: UIColor.black, selected: FFA409, disabled: UIColor.gray)
-            view.setLabelFontWhen(normal: UIFont.systemFont(ofSize: 12), selected: UIFont.boldSystemFont(ofSize: 12), disabled: UIFont.systemFont(ofSize: 12))
+            view.setLabelColorWhen(normal: .black, selected: FFA409, disabled: .gray)
+            view.setLabel(font: .systemFont(ofSize: 12))
```
Comment (younatics) [inline YNDropDownMenu/Example/ViewController.swift:23]: Great, However I have to change menu label font to bold if menu is selected 👍

Comment (zntfdr) [reply YNDropDownMenu/Example/ViewController.swift:23]: Ouch, I've missed that!

Comment (younatics) [reply YNDropDownMenu/Example/ViewController.swift:23]: I will add this method too :)

**Review by younatics — APPROVED** [review]
Great pull request 💯

File: YNDropDownMenu/Example/ViewController.swift (hunk 3 above)
Comment (zntfdr) [reply YNDropDownMenu/Example/ViewController.swift:23]: Awesome, thanks! 
I can write a new PR reversing the fonts in the example if you'd like 😊

Comment (younatics) [reply YNDropDownMenu/Example/ViewController.swift:23]: I committed again :) 
Again, Thanks!

Comment (zntfdr) [reply YNDropDownMenu/Example/ViewController.swift:23]: Great! 😄

**Pull request merged**